### PR TITLE
[Hardware][Ascend] Skip Omni hidden-state cache writes for pooling outputs

### PR DESCRIPTION
**Pull request by Big2Wheel**
## Purpose

Prevent pooling outputs from being written into the Omni hidden-state prefix cache on Ascend, matching the GPU runner's pooling guard.

Qwen3-ForcedAligner produces 5000-wide classification logits, whereas its hidden-state cache is 1024-wide. With Omni prefix caching enabled, the NPU runner attempts an incompatible `index_copy_` and fails. This patch skips Omni tensor-cache writes for pooling models; it does **not** disable KV prefix caching or change generation-model cache writes.

Adds four regression cases (pooling/generation × cache enabled/disabled), using real NPU tensors, and includes them in the existing A2 single-card Buildkite job. This fix is independent of #7544.

## Test Plan

**vLLM Version:** `0.29.0+empty`, official `quay.nju.edu.cn/ascend/vllm-omni:v0.29.0rc1`; torch `2.10.0`, torch-npu `2.10.0.post4`, Transformers `5.14.1`.

**vLLM-Omni Commit:** `92715f3f7b67d7835c08002309402ba08b7b237e` + this patch.

Hardware: Ascend A2 / 910B3 64 GB. Run in a compatible NPU environment with pytest-mock:

```bash
pytest -q tests/platforms/npu/test_pooling_prefix_cache.py
```

To reproduce the original failure, retain the test but restore the runner from the base commit, then run the same command.

## Test Result

- Before: **1 failed, 3 passed**; pooling + cache enabled reproduces the 5000/1024 slice-shape mismatch. After: **4 passed**.
- Combined real-model validation on two A2 NPUs also passed with Qwen3-TTS-12Hz-1.7B-CustomVoice and Qwen3-ForcedAligner-0.6B: HTTP/streaming timestamps, concurrent requests and cancellation recovery, with aligner prefix caching and graph capture enabled. This combined run additionally included #7544 and the separate full-audio alignment fix; it is not standalone streaming evidence for this guard.
- Ruff, formatting, applicable source hooks, YAML/Buildkite schema and whitespace checks passed. Hosted CI has not run locally. Mypy reports nine existing runner errors, also reproduced on clean main; the new test passes typing. The all-hooks bootstrap encountered an unrelated actionlint Go download failure, so no all-hooks-green claim is made.

AI assistance: OpenAI Codex assisted with implementation, regression tests, validation and this description. The human author confirmed review of both the changes and authorship responsibility before submission.

### CODE REVIEW

**Comment by Big2Wheel** [comment]
Closing as superseded by merged #6654, rather than #7544. The shared prefix-cache path now excludes pooling models from Omni hidden-state cache writes, covering the 5000/1024 logits-versus-hidden-state shape mismatch addressed here.

We revalidated Qwen3-TTS-12Hz-1.7B-CustomVoice + Qwen3-ForcedAligner-0.6B on two Ascend 910B3 64 GB (A2) cards using the v0.29.0rc1 image and #7583 head d14dcbfbe905dcb876ee17f6fd1a830ea70d4605 (main base f3f8ebfc), without applying this PR. HTTP/streaming timestamps, concurrency and cancellation recovery passed. This was combined validation with the separate full-audio alignment fix in #7583, not a standalone clean-main E2E claim.

#7583 addresses a different issue (complete waveform delivery for DELTA streaming alignment) and remains open. No further merge of this pooling guard is needed. Thanks!
